polkadot/node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -1558,6 +1565,61 @@ impl Initialized {
 	fn session_is_ancient(&self, session_idx: SessionIndex) -> bool {
 		return session_idx < self.highest_session_seen.saturating_sub(DISPUTE_WINDOW.get() - 1)
 	}
+
+	/// Revisit active non-confirmed disputes after validators have been disabled.
+	/// Unactivates disputes where all raising parties (invalid voters) are now disabled.
+	fn revisit_active_disputes_after_disabling(
+		&mut self,
+		overlay_db: &mut OverlayedBackend<'_, impl Backend>,
+		session: SessionIndex,
+	) -> FatalResult<()> {
+		let mut recent_disputes = overlay_db.load_recent_disputes()?.unwrap_or_default();
+		let mut disputes_to_remove = Vec::new();
+
+		// Create session bounds for efficient iteration
+		let session_start = (session, CandidateHash(Hash::zero()));
+		let session_end = (session + 1, CandidateHash(Hash::zero()));
+
+		for ((dispute_session, candidate_hash), status) in
+			recent_disputes.range(&session_start..&session_end)
+		{
+			debug_assert_eq!(session, *dispute_session);
+			// Only check unconfirmed
+			if status.is_confirmed_concluded() {
+				continue
+			}
+			let Some(votes) = overlay_db.load_candidate_votes(*dispute_session, candidate_hash)?
+			else {
+				continue
+			};
+			// Check if all invalid voters (raising parties) are disabled
+			if !votes.invalid.is_empty() &&
```

Comment:
Can invalid votes contain votes from not only disabled validators?

Reply:
They definitely can. Validators become disabled only after the dispute they raised gets concluded - there could be multiple disputes going on with different conclusion times. It's also possible in theory that backers were malicious/faulty and the block is actually invalid, so honest nodes should vote invalid.

Reply:
Sorry, missed that there is `.all` from disabled